### PR TITLE
[Doc] update wrappable talons topic

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Minor content update to the wrappable talons topic.
Also fixes a broken link.

## Related Issue

Closes PWA-1077

## Acceptance

Any core developer

### Verification Stakeholders

Any core developer

## Test Plan

1. Read changes
2. Verify content is accurate

#### Test scenario(s) for direct fix/feature
Example
-   Verify user is able to apply filters on category page
-   Verify user is able to apply filters on search page

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md [hunk 1]
```diff
@@ -5,23 +5,25 @@ title: Wrappable Peregrine Talons
...
 
 ### Wrapper modules
 
 Talons are React hooks, and React hooks are plain functions.
 Therefore, you can use the function wrapper pattern to intercept talon functions.
 
-The `talon.wrapWith(module)` method is simlar to the [interceptor pattern](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/plugins.html) also used in Magento backend plugins.
-Peregrine will dynamically inject the code from the passed `module` "around" the implementation of a talon, by passing the talon function through the wrapper function before exporting it.
+The `talon.wrapWith(module)` method is similar to the [interceptor pattern](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/plugins.html) used in Magento backend plugins.
+During the build process, the extensibility framework dynamically wraps a talon implentation with code from a wrapper module by passing the talon function through the wrapper module before exporting it.
```
Comment (larsroettig) [inline pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md:17]: The extensibility framework dynamically wraps a talon implementation with code from a wrapper module by passing the talon function through the wrapper module before exporting it during the build process.

**Review by larsroettig — CHANGES_REQUESTED** [review]

File: pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md [hunk 2]
```diff
@@ -5,23 +5,25 @@ title: Wrappable Peregrine Talons
 ## Usage
 
 An extension may need to modify or decorate the behavior of a Peregrine talon.
```
[the remark this answers is not in the document]
Comment (jcalcaben) [reply pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md:7]: this is the correct spelling: https://www.google.com/search?q=define%3Abehavior

File: pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md [hunk 3]
```diff
@@ -5,23 +5,25 @@ title: Wrappable Peregrine Talons
...
-Third-party code may need to log to an external source when certain events occur, or may need to extend the model returned by the talon when a component calls it.
+For example, third-party code may need to log to an external source when certain events occur or it may need to extend the model returned by the talon when a component calls it.
```
[the remark this answers is not in the document]
Comment (jcalcaben) [reply pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md:8]: I'm not sure the comma is required before this "or". The sentence is a list of two items, e.g. "For example, this first case **or** this other case."

File: pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md (hunk 1 above)
Comment (jcalcaben) [reply pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md]: Fixed the mispelled word but kept the ordering because it helps to introduce the context at the beginning of the sentence.

**Review by larsroettig — APPROVED** [review]
:heavy_check_mark:  code review
